client/state/post-types/taxonomies/selectors.js

```diff
@@ -34,3 +35,22 @@ export function getPostTypeTaxonomies( state, siteId, postType ) {
 
 	return values( taxonomies );
 }
+
+/**
+ * Returns the given taxonomy for the given post type on a site, or null if the
+ * taxonomies are not known.
+ *
+ * @param  {Object}  state        Global state tree
+ * @param  {Number}  siteId       Site ID
+ * @param  {String}  postType     Post type
+ * @param  {String}  taxonomyName Taxonomy name
+ * @return {Object?}              Post type taxonomy
+ */
+export function getPostTypeTaxonomy( state, siteId, postType, taxonomyName ) {
+	const taxonomies = getPostTypeTaxonomies( state, siteId, postType );
+	if ( ! taxonomies ) {
+		return null;
+	}
+
+	return find( taxonomies, tax => tax.name === taxonomyName ) || null;
```

Comment:
You may find Lodash's object predicate shorthand (available for many of its functions) to be useful here as an alternative to the callback:

```js
return find( taxonomies, { name: taxonomyName } ) || null;
```